docs/arcadia-users-group/20230228-intro-to-python-1/lesson.md

```diff
@@ -0,0 +1,548 @@
+
+
+```python
+1 + 2 * 3.4 / 5
+```
+
+
+
+
+    2.36
+
+
+
+Python also has some helpful intuitive operations – for example, you can join two strings together with a `+` operator as follows:
+
+
+```python
+'pine' + 'apple' 
+```
+
+
+
+
+    'pineapple'
+
+
+
+Certain operations are not supported by default in Python - for example, you can't add a `str` with an `int`. Running the code below will cause a `TypeError`. Python is quite helpful in explaining errors you might encounter. Read the error message below and see if it makes sense to you.
+
+In general, you should expect to encounter *lots* of these errors, or "bugs", when you start getting into programming. **This is totally normal!**  
+In fact, probably 50% of programming is just squashing bugs.  
+
+If you're ever confused about the source of an error, websites such as [StackOverflow](https://stackoverflow.com/) are great places to look for answers.
+
+
+```python
+3 + '5'
+```
+
+
+    ---------------------------------------------------------------------------
+
+    TypeError                                 Traceback (most recent call last)
+
+    Cell In[5], line 1
+    ----> 1 3 + '5'
+
+
+    TypeError: unsupported operand type(s) for +: 'int' and 'str'
+
+
+> **Fun Fact!**
+>
+> The term "bug" predates modern computers, but there are real examples of actual insects interfering with the function of a computer, such as this ["bug" found at Harvard in 1947](https://education.nationalgeographic.org/resource/worlds-first-computer-bug). 
+
+---
+## 3. Using variables
+
+You assign variables in Python using the `variable = value` syntax. You can perform operations on a mix of hard-coded values and variables interchangeably. This is very useful when you expect to use a variable in many different places in your script - Python keeps track of its value so you don't have to.
+
+
+```python
+a = 1
+b = 6 + a
+c = a * b
+
+a + b + c
+```
+
+
+
+
+    15
+
+
+
+You can assign a different value to a variable later in your code, which replaces the original value. This can be useful, but it's also important to be careful not to overwrite variables you want to keep!
+
+
+```python
+a = 4
+
+a + b
+```
+
+
+
+
+    10
+
+
+
+---
+## 4. Built-in Python functions
+
+In addition to managing variables and values, Python also provides a variety of [built-in functions](https://docs.python.org/3/library/functions.html). Functions are "called" by chaining the function name with parentheses `()`. Parameters of that function can be typed between the parentheses to tell the function what to do.
+
+For example, the `print()` function takes any value and prints it as output.
+
+
+```python
+print('Hello, world!')
+
+print(a + b)
+```
+
+    Hello, world!
+    10
+
+
+We can call a function from within another function call. For example, the `type()` function gives the type of a Python value.
+
+
+```python
+print(type(4))
+print(type('Hello, world!'))
+```
+
+    <class 'int'>
+    <class 'str'>
+
+
+---
```

Comment:
do you use this much when you're programming? do you think it's best practice, or important for learner's mental model? if not it might be useful to take it out

Reply:
I actually use this a lot when dealing with imported data in `pandas` - it helps manage NaNs and situations where columns have mixed data types!